From 87d74183f00fc553c4d5766c8330f65b850bbe22 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Caio=20Rom=C3=A3o?= <caioromao@gmail.com>
Date: Wed, 17 Jun 2009 17:43:01 -0300
Subject: [PATCH 08/10] Making modify() work on Bugzilla 3.2.0.1-1.

Some request parameters must be explicitly set to make it all work.
---
 bugz/bugzilla.py |   24 ++++++++++++++++++------
 1 files changed, 18 insertions(+), 6 deletions(-)

diff --git a/bugz/bugzilla.py b/bugz/bugzilla.py
index 5228bdb..99ac613 100644
--- a/bugz/bugzilla.py
+++ b/bugz/bugzilla.py
@@ -514,7 +514,6 @@ class Bugz:
         if not self.authenticated:
             self.auth()
 
-
         buginfo = Bugz.get(self, bugid)
         if not buginfo:
             return False
@@ -526,9 +525,10 @@ class Bugz:
 
         # copy existing fields
         FIELDS = ('bug_file_loc', 'bug_severity', 'short_desc', 'bug_status',
-                'status_whiteboard', 'keywords',
-                'op_sys', 'priority', 'version', 'target_milestone',
-                'assigned_to', 'rep_platform', 'product', 'component')
+                  'status_whiteboard', 'keywords',
+                  'op_sys', 'priority', 'version', 'target_milestone',
+                  'assigned_to', 'rep_platform', 'product', 'component',
+                  'delta_ts')
 
         FIELDS_MULTI = ('blocked', 'dependson')
 
@@ -559,6 +559,7 @@ class Bugz:
             modified.append(('resolution', qparams['resolution']))
         elif status == 'REOPENED' and status != qparams['bug_status']:
             qparams['knob'] = 'reopen'
+            qparams['bug_status'] = 'REOPENED'
             modified.append(('status', status))
         elif status == 'VERIFIED' and status != qparams['bug_status']:
             qparams['knob'] = 'verified'
@@ -568,7 +569,9 @@ class Bugz:
             modified.append(('status', status))
         elif duplicate:
             qparams['knob'] = 'duplicate'
-            qparams['dup_id'] = duplicate
+            qparams['dup_id'] = str(duplicate)
+            qparams['bug_status'] = 'RESOLVED'
+            qparams['resolution'] = 'DUPLICATE'
             modified.append(('status', 'RESOLVED'))
             modified.append(('resolution', 'DUPLICATE'))
         elif assigned_to:
@@ -645,6 +648,9 @@ class Bugz:
 
         try:
             resp = self.opener.open(req)
+            if duplicate:
+                print qparams
+                print req_params
             return modified
         except:
             return []
@@ -862,8 +868,14 @@ class Bugz:
 
         # TODO: return attachment id and success?
         try:
+            r = resp.read()
             re_success = re.compile(r'<title>Changes Submitted</title>')
-            if re_success.search(resp.read()):
+            if re_success.search(r):
+                return True
+            # Newer Bugzilla versions have a different title
+            re_newattach = re.compile(r'<title>.+ (\d+) .* (\d+)</title>',
+                                      re.I)
+            if re_newattach.search(r):
                 return True
         except:
             pass
-- 
1.6.3.2

